bin.src/genCalibRegistry.py

```diff
@@ -1,148 +1,98 @@
-                print >>sys.stderr, "Warning: Unrecognized file name:", fits
-                raise Exception("Unrecognized file name")
+                m = re.search(r'\w+/([\w-]+)/(\d{4})-(\d{2})-(\d{2})/pfs(\w+)-(\d{6})-(\d)(\w).fits', fits)
+                if not m:
+                    print >>sys.stderr, "Warning: Unrecognized file name:", fits
+                    raise Exception("Unrecognized file name")
+                else:
+                    filterName, year, month, day, version, visit, spectrograph, arm = m.groups()
+                    version = version[0].lower() + version[1:]
             else:
-                filterName, year, month, day, version, visit, spectrograph, arm = m.groups()
-                version = version[0].lower() + version[1:]
-        else:
-            filterName, version, year, month, day, spectrograph, arm = m.groups()
-
-        print "Registering:", fits
-        date = datetime.date(int(year), int(month), int(day))
-        ccd = int(spectrograph) - 1
-        if arm in ("m"):
-            ccd = ccd + 4
-        elif arm in ("r"):
-            ccd += 4
-        elif arm in ("n"):
-            ccd += 8
-
-        if filterName not in rowsPerFilter:
-            rowsPerFilter[filterName] = list()
-        rowsPerFilter[filterName].append(Row(date, version, spectrograph, arm, ccd))
-
-    # Fix up the validStart,validEnd so there are no overlaps
-    for filterName, rows in rowsPerFilter.items():
-        rows.sort(key=lambda row: row.calibDate)
-        validity = datetime.timedelta(opts.validity)
-        valids = collections.OrderedDict([(row.calibDate, [row.calibDate - validity,
-                                                           row.calibDate + validity]) for row in rows])
-        dates = valids.keys()
-        numDates = len(dates)
-        midpoints = [ t1 + (t2 - t1)//2 for t1, t2 in zip(dates[:numDates-1], dates[1:]) ]
-        for i, (date, midpoint) in enumerate(zip(dates[:numDates-1], midpoints)):
-            if valids[date][1] > midpoint:
-                nextDate = dates[i+1]
-                #print "Adjusting: %d %s --> %s : %s vs %s" % (i, date, nextDate, valids[date][1], midpoint)
-                valids[nextDate][0] = midpoint
-                valids[date][1] = midpoint
-        del midpoints
-        del dates
-
-        for row in rows:
-            calibDate = row.calibDate.isoformat()
-            validStart = valids[row.calibDate][0].isoformat()
-            validEnd = valids[row.calibDate][1].isoformat()
-
-            # print "%f --> %f %f" % (calibDate, validStart, validEnd)
-
-            if isSqlite:
+                filterName, version, year, month, day, spectrograph, arm = m.groups()
+
+            print "Registering:", fits
+            date = datetime.date(int(year), int(month), int(day))
+            ccd = int(spectrograph) - 1
+            if arm in ("m"):
+                ccd = ccd + 4
+            elif arm in ("r"):
+                ccd += 4
```

Comment:
Please add a comment that the medium-resolution arm ('m') and the red arm ('r') use the same CCDs and put in one if statement like
```
if arm in ('m', 'r'):
    ccd += 4
```